.github/workflows/release-cycle.yml

```
@@ -0,0 +1,35 @@
+name: Test release cycle
+
+on: [pull_request, push, workflow_dispatch]
+
+env:
+  FORCE_COLOR: 1
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Silence Windows warnings
+        run: |
+          git config core.eol lf
+          git config core.autocrlf input
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3"
+
+      - name: Generate release cycle output
+        run: python -I -bb -X dev -X warn_default_encoding -W error _tools/generate_release_cycle.py
+
+      - name: Check for differences
+        run: |
+          git status
+          git diff
+          test $(git status --porcelain | wc -l) = 0
```

```suggestion
          git add .
          git status
          git diff --staged
          test $(git status --porcelain | wc -l) = 0
```

_This_ should work, as I thought up in my reply to @merwok , since if files are checked out with `CRLF` due to `core.autocrlf true`, once staged to the index there will be no actual changes, since the LF EOL matches what is stored in the repository. (And if this somehow isn't the case, we perhaps _do_ have a potential broader problem we need to address).

---

I think we don't need this after https://github.com/python/devguide/pull/988#discussion_r1035574945?

---

I had also figured out the cause, but determined per the discussion above that this is both a simpler and more robust solution, as it verifies that local users won't see a difference at least after staging.